server/models/express-usage-record.js

```diff
@@ -35,4 +54,84 @@ module.exports = function extendExpressUsageRecord(ExpressUsageRecord) {
     });
   }
   ExpressUsageRecord.recordUsage = recordUsage;
+
+  function dailySummary(callback) {
+    ExpressUsageRecord.find({order: ['lbModelName', 'requestUrl']},
+      function(err, records) {
+        if (err) return callback(err);
+        var bucket = {};
+        for (var r in records) {
+          if (!records.hasOwnProperty(r)) continue;
+          var key = records[r].lbModelName;
+          if (!key) {
+            key = records[r].requestUrl;
+          }
+
+          if (!bucket.hasOwnProperty(key)) bucket[key] = 0;
+          bucket[key] += 1;
+        }
+        callback(null, bucket);
+      }
+    );
+  }
+  ExpressUsageRecord.dailySummary = dailySummary;
+
+  function hourlySummary(modelOrUri, window, callback) {
+    var windowStart = new Date(window);
+    windowStart.setMinutes(0);
+    windowStart.setSeconds(0);
+    windowStart.setMilliseconds(0);
+    windowStart = Number(windowStart);
+
+    var windowEnd = windowStart + 60 * 60 * 1000;
+    var q = {
+      where: {
+        or: [{lbModelName: modelOrUri}, {requestUrl: modelOrUri}],
+        timeStamp: {gt: windowStart, lt: windowEnd}
+      }
+    };
+    ExpressUsageRecord.find(q, function(err, records) {
+      if (err) return callback(err);
+      var bucket = new Array(20);
+      for (var i = 0; i < 20; i++) {
+        bucket[i] = {
+          timeStamp: windowStart + (i * 5) * 60 * 1000,
+          GET: 0,
+          POST: 0,
+          PUT: 0,
+          DELETE: 0
+        };
+      }
+
+      for (var r in records) {
+        if (!records.hasOwnProperty(r)) continue;
+        var ts = records[r].timeStamp;
+        var method = records[r].requestMethod;
+        var key = Math.floor(ts.getMinutes() / 5);
+        bucket[key][method] += 1;
+      }
+      callback(null, bucket);
+    });
+  }
+  ExpressUsageRecord.hourlySummary = hourlySummary;
+
+  function endpointDetail(modelOrUri, window, callback) {
```

Comment:
Same comment here, would you mind using a different name, e.g. `windowStartTime`?